### PR TITLE
Feature/latest python2

### PR DESCRIPTION
**Pull request by gjbex**
EasyConfig for Python 2.7.9:
* patch for missing C type definition for Intel compilers seems no longer necessary
* updated Python packages listed as extensions to latest versions
* Updated OpenSSL dependency to latest versino
EasyConfig for OpenSSL 1.0.1k

### CODE REVIEW

File: easybuild/easyconfigs/p/Python/Python-2.7.9-intel-2015a.eb [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+version = '2.7.9'
+
+homepage = 'http://python.org/'
+description = "Python is a programming language that lets you work more quickly and integrate your systems more effectively."
+
+toolchain = {'name': 'intel', 'version': '2015a'}
+toolchainopts = {'pic': True, 'opt': True, 'optarch': True}
+
+numpyversion = '1.9.1'
+scipyversion = '0.14.1'
```
Comment (wpoely86) [inline easybuild/easyconfigs/p/Python/Python-2.7.9-intel-2015a.eb]: why not update to `0.15`?

File: easybuild/easyconfigs/p/Python/Python-2.7.9-intel-2015a.eb [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        ],
+    }),
+    ('scipy', scipyversion, {
+        'source_urls': [('http://sourceforge.net/projects/scipy/files/scipy/%s' % scipyversion, 'download')],
+    }),
+    ('blist', '1.3.6', {
+        'source_urls': ['http://pypi.python.org/packages/source/b/blist/'],
+    }),
+    ('mpi4py', '1.3.1', {
+        'source_urls': ['https://mpi4py.googlecode.com/files/'],
```
Comment (wpoely86) [inline easybuild/easyconfigs/p/Python/Python-2.7.9-intel-2015a.eb]: moved to `https://bitbucket.org/mpi4py/mpi4py`

**Comment by wpoely86** [comment]
ok to test

**Comment by gjbex** [comment]
scipy 0.15.0 is not mentioned yet on www.scipy.org, hence I hesitated, but I'd be happy to change it if you think it is ok.

**Comment by wpoely86** [comment]
Weird, it's here http://sourceforge.net/projects/scipy/files/scipy/0.15.0/

Haven't tried it myself.

**Comment by gjbex** [comment]
I have it installed on my own Linux instance, hence my surprise that it was not yet announced on www.scipy.org and my hesitation to jump to that version.

**Comment by wpoely86** [comment]
This is gonna clash with #1291

**Comment by gjbex** [comment]
I noticed only (a while) after issuing the pull request that there was one from Kenneth on this.  Perhaps it is simplest to just drop this one.

**Comment by wpoely86** [comment]
If all the versions of the extensions are the same, it's probably best to close this one.

**Comment by boegel** [comment]
@wpoely86: thanks for pointing this out

@gjbex: diff between my Python easyconfig and yours:

```diff
--- easybuild/easyconfigs/p/Python/Python-2.7.9-intel-2015a.eb	2015-01-15 20:13:06.000000000 +0100
+++ Python-2.7.9-intel-2015a.eb	2015-01-15 20:12:46.000000000 +0100
@@ -23,8 +23,6 @@
 #   nice to have an up to date openssl for security reasons
 ]
 
-osdependencies = [('openssl-devel', 'libssl-dev')]
-
 # order is important!
 # Version updated 08/JULY/2014
 exts_list = [
@@ -67,7 +65,7 @@
     ('six', '1.9.0', {
         'source_urls': ['https://pypi.python.org/packages/source/s/six/'],
     }),
-    ('dateutil', '2.2', {
+    ('dateutil', '2.4.0', {
         'source_tmpl': 'python-%(name)s-%(version)s.tar.gz',
         'source_urls': ['http://pypi.python.org/packages/source/p/python-dateutil/'],
     }),
@@ -96,4 +94,6 @@
     }),
 ]
 
+osdependencies = [('openssl-devel', 'libssl-dev')]
+
 moduleclass = 'lang'
```

so, you are using a more recent dateutil, I should too

what are we going to do with the scipy version? scipy v0.15.0 was only 'released' Jan 11th 2015, and it's a 'major' version bump, so I'm included to stick with v0.14.1 (which was released Dec 30th 2014, so it's also quite fresh)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/f6c383d5cc925d9bb571 for a full test report.

**Comment by gjbex** [comment]
Ok for scipy 0.14.1, seems the safer option (in contrast to 0.15.0).

**Comment by boegel** [comment]
Hence, good to go, I'll the conflicts that will appear in #1291.

Thanks @gjbex/@hpcleuven!

**Pull request merged**